### PR TITLE
[Internal] Retrieve trading pairs from PancakeSwap

### LINKED ISSUE #5331: Integrate automated pulling of trading pairs from Uniswap And PancakeSwap
**Is your feature request related to a problem? Please describe.**

We need to enable automatic pull for pairs that are:
- have at least $1M in liquidity

Also this should include ability to add specific pairs manually, when needed. 

Asset information could be pulled from: https://api.trustwallet.com/v1/assets/c714_tBTCB-1DE
 
#### Uniswap: 
List: https://github.com/trustwallet/assets/blob/master/blockchains/ethereum/tokenlist.json
Tokens info: https://info.uniswap.org/tokens
API: https://uniswap.org/docs/v2/API/queries/

#### PancakeSwap: 
List: https://github.com/trustwallet/assets/blob/master/blockchains/smartchain/tokenlist.json
Tokens: https://pancakeswap.info/

### PR DESCRIPTION
**Pull request by optout21**
This PR is being closed due to inactivity. If you wish to continue, please have us reopen the PR before sending your payment.
Do NOT send payments for closed PR, as the fee may by lost!
  (79 vs. 60 hrs)

### CODE REVIEW

**Comment by catenocrypt** [comment]
So far pairs are retireved, but not yet saved.
Sample output:

```
Retrieved & filtered 37 pairs:
pair:  Cake -- WBNB 	 USD 27481386  0x0e09fabb73bd3ade0a17ecc321fd13a19e81ce82 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  WBNB -- BUSD 	 USD 26641646  0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c 0xe9e7cea3dedca5984780bafc599bd69add087d56
pair:  WBNB -- UNI 	 USD 9165536  0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c 0xbf5140a22578168fd562dccf235e5d43a02ce9b1
pair:  ETH -- WBNB 	 USD 7869468  0x2170ed0880ac9a755fd29b2688956bd959f933f8 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  BTCB -- WBNB 	 USD 7673736  0x7130d2a12b9bcbfae4f2634d864a1ee1ce3ead9c 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  USDT -- WBNB 	 USD 7034397  0x55d398326f99059ff775485246999027b3197955 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  USDT -- BUSD 	 USD 6673868  0x55d398326f99059ff775485246999027b3197955 0xe9e7cea3dedca5984780bafc599bd69add087d56
pair:  WBNB -- LINK 	 USD 6028236  0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c 0xf8a0bf9cf54bb92f17374d9e9a321e6a111a51bd
pair:  DAI -- BUSD 	 USD 4330725  0x1af3f329e8be154074d8769d1ffa4ee058b1dbc3 0xe9e7cea3dedca5984780bafc599bd69add087d56
pair:  DOT -- WBNB 	 USD 4240458  0x7083609fce4d1d8dc0c979aab8c869ea2c873402 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  BAND -- WBNB 	 USD 4145862  0xad6caeb32cd2c308980a548bd0bc5aa4306c6c18 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  EOS -- WBNB 	 USD 3966043  0x56b6fb708fc5732dec1afc8d8556423a2edccbd6 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  BDO -- BUSD 	 USD 3737111  0x190b589cf9fb8ddeabbfeae36a813ffb2a702454 0xe9e7cea3dedca5984780bafc599bd69add087d56
pair:  LTC -- WBNB 	 USD 3697565  0x4338665cbb7b2485a8855a139b75d5e34ab0db94 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  VAI -- BUSD 	 USD 3279041  0x4bd17003473389a42daf6a0a729f6fdb328bbbd7 0xe9e7cea3dedca5984780bafc599bd69add087d56
pair:  BSCX -- BUSD 	 USD 3181476  0x5ac52ee5b2a633895292ff6d8a89bb9190451587 0xe9e7cea3dedca5984780bafc599bd69add087d56
pair:  TWT -- WBNB 	 USD 2796377  0x4b0f1812e5df2a09796481ff14017e6005508003 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  USDC -- BUSD 	 USD 2661173  0x8ac76a51cc950d9822d68b83fe1ad97b32cd580d 0xe9e7cea3dedca5984780bafc599bd69add087d56
pair:  ETH -- bALBT 	 USD 2459502  0x2170ed0880ac9a755fd29b2688956bd959f933f8 0x72faa679e1008ad8382959ff48e392042a8b06f7
pair:  SXP -- WBNB 	 USD 2375565  0x47bead2563dcbf3bf2c9407fea4dc236faba485a 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  XRP -- WBNB 	 USD 1876892  0x1d2f0da169ceb9fc7b3144628db156f3f6c60dbe 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  WBNB -- XVS 	 USD 1765961  0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c 0xcf6bb5389c92bdda8a3747ddb454cb7a64626c63
pair:  ADA -- WBNB 	 USD 1721327  0x3ee2200efb3400fabb9aacf31297cbdd1d435d47 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  ALPHA -- WBNB 	 USD 1699507  0xa1faa113cbe53436df28ff0aee54275c13b40975 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  sBDO -- BUSD 	 USD 1655274  0x0d9319565be7f53cefe84ad201be3f40feae2740 0xe9e7cea3dedca5984780bafc599bd69add087d56
pair:  ATOM -- WBNB 	 USD 1621770  0x0eb3a705fc54725037cc9e008bdede697f62f335 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  YFI -- WBNB 	 USD 1542480  0x88f1a5ae2a3bf98aeaf342d26b30a79438c9142e 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  FIL -- WBNB 	 USD 1502308  0x0d8ce2a99bb6e3b7db580ed848240e4a0f9ae153 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  INJ -- WBNB 	 USD 1456492  0xa2b726b1145a4773f68593cf171187d8ebe4d495 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  bALBT -- WBNB 	 USD 1363821  0x72faa679e1008ad8382959ff48e392042a8b06f7 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  BTCB -- BUSD 	 USD 1353815  0x7130d2a12b9bcbfae4f2634d864a1ee1ce3ead9c 0xe9e7cea3dedca5984780bafc599bd69add087d56
pair:  BSCX -- WBNB 	 USD 1225265  0x5ac52ee5b2a633895292ff6d8a89bb9190451587 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  CTK -- WBNB 	 USD 1207007  0xa8c2b8eec3d368c0253ad3dae65a5f2bbb89c929 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  BCH -- WBNB 	 USD 1137378  0x8ff795a6f4d97e7887c79bea79aba5cc76444adf 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  DAI -- USDT 	 USD 1018872  0x1af3f329e8be154074d8769d1ffa4ee058b1dbc3 0x55d398326f99059ff775485246999027b3197955
pair:  DITTO -- WBNB 	 USD 1010207  0x233d91a0713155003fc4dce0afa871b508b3b715 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
pair:  UNFI -- WBNB 	 USD 1000058  0x728c5bac3c3e370e372fc4671f9ef6916b814d8b 0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c
```

**Comment by catenocrypt** [comment]
Also added trading pair retrieve from Uniswap:

```
Retrieved & filtered 131 pairs:
pair:  USDC -- WETH 	 USD 254685306 	 0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2
pair:  WBTC -- WETH 	 USD 216758471 	 0x2260fac5e5542a773aa44fbcfedf7c193bc2c599 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2
pair:  WETH -- USDT 	 USD 214093901 	 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2 0xdac17f958d2ee523a2206206994597c13d831ec7
...
pair:  DAI -- HEZ 	 USD 1734818 	 0x6b175474e89094c44da98b954eedeac495271d0f 0xeef9f339514298c6a857efcfc1a762af84438dee
pair:  SUSHI -- WETH 	 USD 1730350 	 0x6b3595068778dd592e39a122f4f5a5cf09c90fe2 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2
pair:  SHARE -- WETH 	 USD 1700972 	 0x39795344cbcc76cc3fb94b9d1b15c23c2070c66d 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2
```

**Comment by catenocrypt** [comment]
I attach here the original `smartchain\tokenlist.json`, it's updated version as by this PR, and the diff.
Also the similar 3 files for Ethereum.
Lists in the files are sorted, for better diff'ing.

[diff.zip](https://github.com/trustwallet/assets/files/5869255/diff.zip)

**Comment by catenocrypt** [comment]
New tokens added to pairs now have named adjusted from TW API, e.g. 'Binance-Peg Cosmos' instead of 'Cosmos Token'.

Resulting changes:
```
Token name adjusted: 'ChainLink Token' -> 'Binance-Peg ChainLink'
Token name adjusted: 'Polkadot Token' -> 'Binance-Peg Polkadot'
Token name adjusted: 'Band Protocol Token' -> 'Binance-Peg Band Protocol'
Token name adjusted: 'EOS Token' -> 'Binance-Peg EOS'
Token name adjusted: 'Cardano Token' -> 'Binance-Peg Cardano'
Token name adjusted: 'Litecoin Token' -> 'Binance-Peg Litecoin'
Token name adjusted: 'XRP Token' -> 'Binance-Peg XRP'
Token name adjusted: 'yearn.finance' -> 'Binance-Peg yearn.finance'
Token name adjusted: 'Bitcoin Cash Token' -> 'Binance-Peg Bitcoin Cash'
Token name adjusted: 'Cosmos Token' -> 'Binance-Peg Cosmos'
Token name adjusted: 'Filecoin' -> 'Binance-Peg Filecoin'
Token name adjusted: 'Tezos Token' -> 'Binance-Peg Tezos'
```

File: script/config.ts [hunk 1]
```diff
@@ -6,4 +6,15 @@ export const imageMaxLogoSizeKb =  100;
...
 export const binanceUrlTokenAssets =  "https://explorer.binance.org/api/v1/assets?page=1&rows=1000";
 export const binanceDexURL = 'https://dex-atlantic.binance.org/api'
-export const assetsURL = 'https://raw.githubusercontent.com/trustwallet/assets/master'
\ No newline at end of file
+export const assetsURL = 'https://raw.githubusercontent.com/trustwallet/assets/master'
+
+export const PancakeSwap_TradingPairsUrl = "https://api.bscgraph.org/subgraphs/name/wowswap";
+export const PancakeSwap_TradingPairsQuery = "query pairs {\\n  pairs(first: 400, orderBy: reserveUSD, orderDirection: desc) {\\n id\\n reserveUSD\\n trackedReserveETH\\n volumeUSD\\n txCount \\n   untrackedVolumeUSD\\n __typename\\n token0 {\\n id\\n symbol\\n name\\n decimals\\n __typename\\n }\\n token1 {\\n id\\n symbol\\n name\\n decimals\\n __typename\\n }\\n }\\n}\\n";
```
Comment (vikmeup) [inline script/config.ts:12]: can you use formatter for this query? so we can easily adjust

Comment (catenocrypt) [reply script/config.ts]: Good idea, done

**Pull request merged**